compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1645,17 +1645,34 @@ impl<'context> Elaborator<'context> {
 
         let name = &alias.type_alias_def.name;
         let visibility = alias.type_alias_def.visibility;
-        let location = alias.type_alias_def.typ.location;
+        let location = alias.type_alias_def.location;
 
         let generics = self.add_generics(&alias.type_alias_def.generics);
         self.current_item = Some(DependencyId::Alias(alias_id));
-        let typ = self.resolve_type(alias.type_alias_def.typ);
+
+        let (typ, num_expr) = if let Some(num_type) = alias.type_alias_def.numeric_type {
+            let num_type = self.resolve_type(num_type);
+            let kind = Kind::numeric(num_type);
+            let num_expr = if let UnresolvedTypeData::Expression(expr) =
```

Comment:
This looks to be the issue with the identity alias. This is a pretty fragile check it seems as we can avoid it by just ensuring that the last step is an expression and it doesn't look deeper?

`type Quadruple<let N: u32>: u32 = Double<N> + Double<N>;`